src/main/java/at/hannibal2/skyhanni/features/garden/farming/FarmingArmorDrops.kt

```diff
@@ -88,9 +93,50 @@ class FarmingArmorDrops {
     }
 
     private fun checkArmor() {
-        val armorPieces = InventoryUtils.getArmor()
+        val armorPieces = InventoryUtils.getArmor(1500)
             .mapNotNull { it?.getInternalName_old() }
             .count { armorPattern.matcher(it).matches() }
         hasArmor = armorPieces > 1
     }
+
+    @SubscribeEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        try {
+            armorDropJson = event.getConstant("ArmorDrops") ?: error("ArmorDrops not found in repo")
+
+        } catch (e: Exception) {
+            e.printStackTrace()
+            LorenzUtils.error("error in RepositoryReloadEvent")
+        }
+    }
+
+    companion object {
+        var armorDropJson = JsonObject()
+        private var currentArmorDropChance = 0.0
+        private var lastCalculationTime = SimpleTimeMark.farPast()
+
+        fun getDropsPerHour(crop: CropType?): Double {
+            if (crop == null) return 0.0
+
+            if (lastCalculationTime.passedSince() > 5.seconds) {
+                lastCalculationTime = SimpleTimeMark.now()
+
+                val armorDropName = crop.specialDropType
+                val specialDropObject = armorDropJson[armorDropName].asJsonObject
+                val armorName = specialDropObject["armor_type"].asString
+                val pieceCount = InventoryUtils.getArmor(1500)
+                    .mapNotNull { it?.getInternalName_old() }
+                    .count { it.contains(armorName) || it.contains("FERMENTO") }
+
+
+                val dropRates = specialDropObject["chance"].asJsonArray
+                var dropRate = 0.0
+                if (pieceCount > 0 && dropRates.size() >= pieceCount) {
+                    dropRate = dropRates[pieceCount - 1].asDouble
+                }
+                currentArmorDropChance = (dropRate * 60 * 60.0) / 100
```

Comment:
There are absolutely no checks on this JSON handling. Please move the json parsing into the event (as per my last review comment), and also consider using GSONs object mapping feature, instead of manually parsing it.